src/stirling/source_connectors/socket_tracer/uprobe_manager.cc

```diff
@@ -886,8 +886,28 @@ int UProbeManager::DeployGoUProbes(const absl::flat_hash_set<md::UPID>& pids) {
           binary, dwarf_reader_status.msg());
       continue;
     }
+
+    auto build_info_s = ReadGoBuildInfo(elf_reader.get());
+    obj_tools::BuildInfo build_info;
+    std::string go_version;
+    if (build_info_s.ok()) {
+      auto& build_info_pair = build_info_s.ValueOrDie();
+      go_version = build_info_pair.first;
+      build_info = std::move(build_info_pair.second);
+    } else {
+      VLOG(1) << absl::Substitute(
+          "Failed to read build info from binary $0. Likely a Go binary built with an older "
+          "version of Go without module support (pre 1.12)."
+          "Message = $1",
+          binary, build_info_s.status().msg());
```

Comment:
Should we be explicitly skipping/returning in this case?

Reply:
I updated this in https://github.com/pixie-io/pixie/commit/f68e5f158450d3b17627a68b771193dd711c0c92. However, this breaks a test case that traces a Go <= 1.11 binary and why I filed #2210.

My plan is to fix that issue and then we can rebase this change once it's done.

Reply:
https://github.com/pixie-io/pixie/pull/2225 will address this.